tidepool_data_science_metrics/cgm/cgm.py

```diff
@@ -1,104 +1,196 @@
     return round(gmi, round_val)
 
 
 def percent_values_by_range(
-    bg_values, lower_threshold: int, upper_threshold: int, round_val=2
-):
+    bg_array: "np.ndarray[np.int64]",
+    lower_threshold: int == 1,
+    upper_threshold: int == 1000,
+    round_val: int = 2,
+) -> np.float64:
     """
-    Calculate the percent of values that match has a bg within the lower and upper threshold.
+    Calculate the percent of bg values are within the lower and upper thresholds.
     The lower and upper values will be included in the range to calculate on.
 
     Parameters
     ----------
-    bg_values : ndarray
+    bg_array : ndarray
         1D array containing data with `int` type.
     lower_threshold : int
         The the lower value in the range to calculate on.
     upper_threshold : int
         zThe the upper value in the range to calculate on.
-    round_val : int
+    round_val : int, optional
         The number of digits to round the result to.
 
     Returns
     -------
     int
         The percent value by range.
     """
-
+    _validate_bg(bg_array)
     calc_low_thresh, calc_upper_thresh = _validate_input(
         lower_threshold, upper_threshold
     )
     results = round(
         np.where(
-            (bg_values <= calc_upper_thresh) & (bg_values >= calc_low_thresh), 1, 0
+            (bg_array <= calc_upper_thresh) & (bg_array >= calc_low_thresh), 1, 0
         ).sum()
-        / bg_values.size
+        / bg_array.size
         * 100,
         round_val,
     )
     return results
 
 
-def percent_time_in_range(
-    bg_values, lower_threshold: int, upper_threshold: int, round_val=2, time_delta=5
-):
+def percent_time_in_range_70_180(
+    bg_array: "np.ndarray[np.int64]", round_val: int = 2
+) -> np.float64:
+    """
+    Calculate the percent of values with a blood glucose value between 70 and 180.
 
-    calc_low_thresh, calc_upper_thresh = _validate_input(
-        lower_threshold, upper_threshold
+    Parameters
+    ----------
+    bg_array : ndarray
+        1D array containing data with `int` type.
+    round_val : int, optional
+        The number of digits to round the result to.
+
+    Returns
+    -------
+    int
+        The percent value with in the range between 70 and 180.
+    """
+    return percent_values_by_range(
+        bg_array, lower_threshold=70, upper_threshold=180, round_val=round_val
+    )
+
+
+def percent_time_above_180(
+    bg_array: "np.ndarray[np.int64]", round_val: int = 2
+) -> np.float64:
+    """
+    Calculate the percent of values with a blood glucose above 180.
+
+    Parameters
+    ----------
+    bg_array : ndarray
+        1D array containing data with `int` type.
+    round_val : int, optional
+        The number of digits to round the result to.
+
+    Returns
+    -------
+    int
+        The percent values above 180.
+    """
+    return percent_values_by_range(
+        bg_array, lower_threshold=180, upper_threshold=1000, round_val=round_val
```

Comment:
this needs to be 180.5 instead of 180, as it is the percent of values > 180.

Reply:
going to use >